File: News_Service_WebFlux/src/main/java/com/api/AdvancedServiceWebFlux/dto/common/BaseSuccessResponse.java
package com.api.AdvancedServiceWebFlux.dto.common;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@EqualsAndHashCode
public class BaseSuccessResponse {

    private final Integer statusCode;

    private final Boolean success;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    private List<Integer> codes;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    private Long id;

    public BaseSuccessResponse() {
        statusCode = 0;
        success = true;
    }

    public BaseSuccessResponse(int statusCode) {
        this.statusCode = statusCode;
        success = true;
    }

    public BaseSuccessResponse(int statusCode, boolean success) {
        this.statusCode = statusCode;
        this.success = success;
    }
}

File: News_Service_Kotlin/src/main/kotlin/com/dunice/advancedServiceKotlin/dto/common/BaseSuccessResponse.kt
package com.dunice.advancedServiceKotlin.dto.common

import com.fasterxml.jackson.annotation.JsonInclude

open class BaseSuccessResponse(
    val statusCode: Int = 0,
    val success: Boolean = true,
    @JsonInclude(JsonInclude.Include.NON_NULL)
    var codes: List<Int?> = emptyList(),
    @JvmField
    @JsonInclude(JsonInclude.Include.NON_NULL)
    var id: Long? = null
) {
// Custom equals and hashCode if
}